bundles/org.openhab.binding.openthermgateway/src/main/java/org/openhab/binding/openthermgateway/internal/Message.java

```diff
@@ -140,7 +130,7 @@ public boolean overrides(@Nullable Message other) {
         // thermostat, and it's ID is equal to the previous message, then this is an
         // override sent by the OpenTherm Gateway
         return other != null && this.getID() == other.getID()
-                && ("R".equals(this.getCode()) || "A".equals(this.getCode()));
+                && (CodeType.R.equals(this.getCode()) || CodeType.A.equals(this.getCode()));
```

Comment:
Enum comparisons are one of the few instances in which reference comparing is safe.
```suggestion
                && (this.code == CodeType.R || this.code == CodeType.A);
```